models/aether_lat-lon/aether_to_dart.f90

```diff
@@ -0,0 +1,472 @@
+! Converts Aether restart files to a netCDF file
+
+use types_mod, only : r4, MISSING_I, vtablenamelength
+
+use time_manager_mod, only: time_type
+
+use utilities_mod, only :                                    &
+    finalize_utilities, error_handler, E_ERR, E_MSG, E_WARN, &
+    initialize_utilities, do_output
+
+use default_model_mod, only : write_model_time
+
+use transform_state_mod, only :                              &
+    static_init_blocks, aether_name_to_dart,                 &
+    nghost, open_block_file, aether_restart_dirname,         &
+    VT_ORIGININDX, VT_VARNAMEINDX, nvar_neutral,  nvar_ion,  &
+    nx_per_block, ny_per_block, nz_per_block,                &
+    nblocks_lon, nblocks_lat, variables,                     &
+    lats, levs, lons, debug, state_time,                     &
+    block_file_name, nlat, nlon, nlev, purge_chars
+
+use netcdf_utilities_mod, only :                                  &
+    nc_create_file, nc_close_file,                                &
+    nc_begin_define_mode, nc_end_define_mode,                     &
+    nc_define_dimension,                                          &
+    nc_add_global_attribute, nc_add_global_creation_time,         &
+    nc_get_attribute_from_variable, nc_add_attribute_to_variable, &
+    nc_define_real_variable, nc_define_real_scalar,               &
+    nc_get_variable, nc_put_variable,                             &
+    nc_synchronize_file
+
+implicit none
+
+!----------------------------------------------------------------------
+! global storage
+!----------------------------------------------------------------------
+
+integer :: member = MISSING_I, &
+           num_args, ncid
+character(len=3)   :: char_mem
+character(len=31)  :: filter_io_root = 'filter_input'
+character(len=64)  :: filter_io_file = ''
+character(len=512) :: error_string_1, error_string_2
+character(len=31),  parameter :: progname = 'aether_to_dart'
+character(len=256), parameter :: source   = 'aether_lat-lon/aether_to_dart.f90'
+
+character(len=4), parameter :: LEV_DIM_NAME = 'alt'
+character(len=4), parameter :: LAT_DIM_NAME = 'lat'
+character(len=4), parameter :: LON_DIM_NAME = 'lon'
+character(len=4), parameter :: TIME_DIM_NAME = 'time'
+
+character(len=4), parameter :: LEV_VAR_NAME = 'alt'
+character(len=4), parameter :: LAT_VAR_NAME = 'lat'
+character(len=4), parameter :: LON_VAR_NAME = 'lon'
+character(len=4), parameter :: TIME_VAR_NAME = 'time'
+
+!======================================================================
+
+call initialize_utilities(progname)
+
+!----------------------------------------------------------------------
+! Get the ensemble member
+!----------------------------------------------------------------------
+num_args = command_argument_count()
+if (num_args == 0) then
+   write(error_string_1,*) 'Usage: ./aether_to_dart member_number (0-based)' 
+   call error_handler(E_ERR, progname, error_string_1)
+endif
```

Comment:
I guess whoever runs this would need to write a loop to generate all filter restarts one by one. Any reason why not generate all members internally? Is there a scenario where a user doesn't need to create all members?

Reply:
We needed working code as soon as possible and a single member program was simpler to write.
It's possible to run multiple aether_to_darts in parallel, so it shouldn't be much slower than
having aether_to_dart handle all members at once.
There are testing and development scenarios which only need 1 member, and perfect_model_obs.